packages/contracts-bedrock/src/L1/StandardValidator.sol

```diff
@@ -0,0 +1,271 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Libraries
+import { GameType, Claim, GameTypes, Hash } from "src/dispute/lib/Types.sol";
+import { Duration } from "src/dispute/lib/LibUDT.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+
+// Interfaces
+import { ISystemConfig } from "interfaces/L1/ISystemConfig.sol";
+import { IAnchorStateRegistry } from "interfaces/dispute/IAnchorStateRegistry.sol";
+import { IDisputeGameFactory } from "interfaces/dispute/IDisputeGameFactory.sol";
+import { IL1CrossDomainMessenger } from "interfaces/L1/IL1CrossDomainMessenger.sol";
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+import { IOptimismMintableERC20Factory } from "interfaces/universal/IOptimismMintableERC20Factory.sol";
+import { IL1StandardBridge } from "interfaces/L1/IL1StandardBridge.sol";
+import { IL1ERC721Bridge } from "interfaces/L1/IL1ERC721Bridge.sol";
+import { IPermissionedDisputeGame } from "interfaces/dispute/IPermissionedDisputeGame.sol";
+import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+import { IDelayedWETH } from "interfaces/dispute/IDelayedWETH.sol";
+import { ISemver } from "interfaces/universal/ISemver.sol";
+import { IResourceMetering } from "interfaces/L1/IResourceMetering.sol";
+import { IOptimismPortal2 } from "interfaces/L1/IOptimismPortal2.sol";
+import { IPreimageOracle } from "interfaces/cannon/IPreimageOracle.sol";
+
+contract StandardValidatorV180 {
+    ISuperchainConfig internal superchainConfig;
+    address internal l1PAOMultisig;
+    address internal mips;
+
+    // Semver strings for our contracts. Pulled from the Superchain Registry.
+
+    string internal constant systemConfigVersion = "2.3.0";
+    string internal constant permissionedDisputeGameVersion = "1.3.1";
+    string internal constant mipsVersion = "1.2.1";
+    string internal constant optimismPortalVersion = "3.10.0";
+    string internal constant anchorStateRegistryVersion = "2.0.0";
+    string internal constant delayedWETHVersion = "1.1.0";
+    string internal constant disputeGameFactoryVersion = "1.0.0";
+    string internal constant l1CrossDomainMessengerVersion = "2.3.0";
+    string internal constant l1ERC721BridgeVersion = "2.1.0";
+    string internal constant l1StandardBridgeVersion = "2.1.0";
+    string internal constant optimismMintableERC20FactoryVersion = "1.9.0";
+    string internal constant preimageOracleVersion = "1.1.2";
+
+    /// Bytecode hashes of our contracts. Does not include contracts that contain immutables. Pulled from
+    /// the Superchain Registry.
+
+    bytes32 internal constant systemConfigHash = 0x165dece6291f3a071e602c804f4a3e89977a71ec866f46c85eb21448f22f6f9a;
```

Comment:
We should delete the version checks and hash checks and just replace them with implementation address checks. We can do `proxyAdmin.getProxyImplementation()` to get the implementation addresses onchain. These hashes also seem incorrect to me and I'm surprised the tests pass, because here we have the same system config and ASR hashes, but those shouldn't match. Similarly the `permissionedDisputeGameHash` and `delayedWETHHash` hashes match, etc for others

Reply:
done